tests/CMakeLists.txt

```diff
@@ -15,15 +15,25 @@
 #
 #######################
 
+include(add_atsplugin)
+
 function(ADD_AUTEST_PLUGIN _NAME)
   add_library(${_NAME} MODULE ${ARGN})
+  if(LINK_PLUGINS AND BUILD_SHARED_LIBS)
+    target_link_libraries(${_NAME} PRIVATE ts::tsapi ts::tsutil)
+  else()
+    target_include_directories(
+      ${_NAME} PRIVATE "$<TARGET_PROPERTY:libswoc::libswoc,INCLUDE_DIRECTORIES>"
+                       "$<TARGET_PROPERTY:yaml-cpp::yaml-cpp,INCLUDE_DIRECTORIES>"
+    )
+  endif()
   set_target_properties(
     ${_NAME}
     PROPERTIES LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/.libs"
                PREFIX ""
                SUFFIX ".so"
   )
-  target_link_libraries(${_NAME} PRIVATE ts::tsapi ts::tsutil)
+  remove_definitions(-DATS_BUILD) # remove the ATS_BUILD define for plugins to build without issue
```

Comment:
Let's rephrase this comment to explain why we have to remove this so plugins build correctly. As I read this, I myself am not sure why undefining ATS_BUILD would be required for plugin builds. :)

Reply:
I copied this line from `add_atsplugin.cmake`.  I don't think it actually does anything, so I'll remove it.